x-pack/plugins/security_solution/server/lib/sourcerer/routes/index.ts

```diff
@@ -58,8 +59,9 @@ export const createSourcererDataViewRoute = (
           allDataViews.find((dv) => dv.id === dataViewId) ?? null;
 
         const { patternList } = request.body;
-        const patternListAsTitle = patternList.sort().join();
+        const patternListAsTitle = ensurePatternFormat(patternList).join();
         const siemDataViewTitle = siemDataView ? siemDataView.title.split(',').sort().join() : '';
```

Comment:
I think you need to do the `ensurePatternFormat` here as well because they are compared below.
```suggestion
        const siemDataViewTitle = siemDataView ? ensurePatternFormat(siemDataView.title.split(',')).join() : '';
```